bin/desi_qso_mgii_afterburner

```diff
@@ -0,0 +1,250 @@
+                             "restricted (objects which are identify as QSO by the mgII afterburner)" +
+                             " / all (All objects which are tested by the mgII fitter)" +
+                             " --> To have 500 objects in the ouput file: set --target_selection all_targets --save_target all")
+
+    parser.add_argument("--lambda_width", type=str, required=False, default=250,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--max_sigma", type=str, required=False, default=200,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_sigma", type=str, required=False, default=10,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_deltachi2", type=str, required=False, default=16,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_signifiance_A", type=str, required=False, default=3,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_A", type=str, required=False, default=0.0,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+
+    return parser.parse_args()
+
+
+def select_targets_with_mgii_fitter(zbest, fibermap, index_sorted, sel_to_mgii, spectra_name, zbest_name, param_mgii_fitter, DESI_TARGET, save_target):
+    """
+    Run QuasarNet to the object with index_to_QN == True from spectra_name.
+    Then, Re-Run RedRock for the targetids which are selected by QN as a QSO.
+    Args:
+        zbest: fitsio hdu 'ZBEST' from zbest file
+        fibermap:  fitsio hdu 'FIBERMAP' from zbest file
+        index_sorted (int array): index to have fibermap[index_sorted] in the same order than zbest
+                      it is build with the TARGETID increasing
+        sel_to_mgii (bool array): size 500. Select on which objects mgii will be apply (index based on zbest)
+        spectra_name /zbest_name (str): The name of the spectra / associated zbest file
+        param_mgii_fitter (dict): contains info for the MgII fitter as lambda_width, max_sigma
+                                  min_sigma, min_deltachi2, min_signifiance_A, min_A
+        DESI_TARGET (str): name of DESI_TARGET for the wanted version of the target selection
+        save_target (str) : restricted (save only IS_QSO_MGII==true targets) / all (save all the sample)
+    Returns:
+        QSO_sel (pandas dataframe): contains all the information useful to build the QSO cat
+    """
+
+    # just to speed up the process and don't go in mgii_fitter function for nothing..
+    if sel_to_mgii.sum() == 0:
+        QSO_sel = pd.DataFrame()
```

Comment:
Maintainability question: how committed are you to using pandas DataFrame?  It is a perfectly reasonable data object that happens to be in our dependency stack, but elsewhere we have standardized on astropy Tables and numpy named arrays as our supported table-like objects and prefer to not use DataFrames to minimize the number of slightly different syntaxes to keep in our head.  In this case I think some of the DataFrame/FITS work arounds you have to do for multi-dimensional columns might already be natively handled by astropy Tables .  Is there some specific reason to use DataFrames to get some convenience or performance that isn't provided by astropy Tables or numpy arrays, or was it just a default that you are familiar with?

I don't want to make you undo DataFrames just for the sake of reorganizing code, but if these every become problematic for another reason (like interfacing with fits...) we'd probably want to standardize then.

Reply:
I had very bad experiences with astropy tables and fits opening with astropy few years ago. In addition, I was more familiar with pandas dataframe. I used it initially to build a catalog, pandas have great features for matching / unique / concatenation / visualization / ect ...

I agree the work around DataFrame/FITS is not great but I already implemented kind of similar interface so it was ok. 

If there is any problem in the future, I will modify it using `astropy.Table `